Cascade/Cache/Cache.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+
+
+class Cache {
+    static let shared = Stack<Int, Fallback>()
+}
+
+
+private struct StackConstant {
+    static let archiveKey = "kCascadeStackArchiveKey"
+}
+
+
+class Stack<Key: Hashable, Value: Codable> {
+    private let encoder = JSONEncoder()
+    private let decoder = JSONDecoder()
+    private let defaults = UserDefaults.standard
+    private var values: [Key: [Data]] = [:]
+
+
+    init() {
+        restoreValues()
+    }
+
+
+    // MARK: Public methods
+
+    func value(for key: Key, _ block:@escaping (Value) -> Void) -> Bool {
+        guard let values = self.values[key], !values.isEmpty else {
+            return false
+        }
+
+        values.forEach {
+            do {
+                let value = try decoder.decode(Value.self, from: $0)
+                block(value)
+            } catch {
+                fatalError("Unable to decode data \(String(data: $0, encoding: .utf8) ?? "unknown")")
```

Comment:
Since this is a library this should or propagate the exception to the client itself or returning an optional